src/platform/intel/ace/platform.c

```diff
@@ -86,16 +86,15 @@ static struct pm_notifier pm_state_notifier = {
 /* Runs on the primary core only */
 int platform_init(struct sof *sof)
 {
-	int ret;
+	int ret = 0;
 
 	trace_point(TRACE_BOOT_PLATFORM_CLOCK);
 	platform_clock_init(sof);
-	kcps_budget_init();
 
 #if CONFIG_KCPS_DYNAMIC_CLOCK_CONTROL
+	kcps_budget_init();
+
 	ret = core_kcps_adjust(cpu_get_id(), PRIMARY_CORE_BASE_CPS_USAGE);
-#else
-	ret = core_kcps_adjust(cpu_get_id(), CLK_MAX_CPU_HZ / 1000);
 #endif
 	if (ret < 0)
 		return ret;
```

Comment:
please, move this check under `#ifdef` and remove `ret` initialisation

Reply:
Done in today's PR revision.